### PR TITLE
Add Language Switcher

### PR DESCRIPTION
**Pull request by derekblank**
Related to https://github.com/Automattic/studio/issues/25

## Proposed Changes

**(WIP)** Adds Language Picker to User Settings modal to allow user to select Studio UI language

Screenshot 2024-05-01 at 2 48 29 pm

## Testing Instructions

TBD

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

**Comment by derekblank** [comment]
Updating with some notes, findings, and next steps:

- [x] 🎨 The `<LanguagePicker />` [onChange](https://github.com/Automattic/studio/pull/72/files#diff-e7ec7243cd5c365101044cf8609b76a1cfa14916b629b9e192668ce53e651cdbR34) event will need to handle the locale change.
- [x] 💾 The updated preferred locale string should persist in some form of local storage across sessions.
- [x] 🔍 **Investigate**: `<LanguagePicker />` events exist on the renderer thread, while `localeData` is [loaded on the main thread](https://github.com/Automattic/studio/blob/05fa8f2223e74c1b5b6631b2cba0092e1f3a43a2/src/index.ts#L167-L168) (before renderer is invoked) -- therefore, `localStorage` can't be used to persist user locale changes (without modifying one or other other).
- [x] 🔍 **Investigate**: [saveUserData](https://github.com/Automattic/studio/blob/1710b273a833b4388f5b27a9fc166c234be6f0ee/src/storage/user-data.ts#L67) may be the best place to save and load any user-preferred locale string changes.
- [ ] 🔍 **Investigate**: Currently, the JSON translation files are [loaded synchronously](https://github.com/Automattic/studio/blob/1710b273a833b4388f5b27a9fc166c234be6f0ee/src/lib/locale.ts#L42) using `require`, and would need an app restart to load the new locale data. Potentially, the JSON translation files could be loaded async without the need to restart Studio to change the Studio UI language/locale.
- [x] 🏁 All of the above changes should only apply to the Studio app UI itself (and not any Studio-created sites), minimizing the footprint of changes.

File: src/ipc-handlers.ts [hunk 1]
```diff
@@ -542,6 +542,14 @@ export async function saveOnboarding(
 	} );
 }
 
+export async function saveUserLocale( _event: IpcMainInvokeEvent, userLocale: string ) {
+	const userData = await loadUserData();
+	await saveUserData( {
+		...userData,
+		userLocale,
+	} );
```
Comment (derekblank) [inline src/ipc-handlers.ts:550]: I think this approach will be a bit limiting until userData could be saved and loaded across both the main and renderer threads.

**Comment by nguyenquanghoang** [comment]
@derekblank Amazing, This is a feature I have been looking forward to.  🙏 Hope Vietnamese available

**Comment by nguyenquanghoang** [comment]
@wojtekn This is awesome

File: src/components/language-picker.tsx
Comment (nguyenquanghoang) [inline src/components/language-picker.tsx:1]: Can you add Vietnamese language?

**Comment by sejas** [comment]
- Closing in favor of https://github.com/Automattic/studio/pull/476

**Pull request closed without merging**